resources/manage.rb

```diff
@@ -17,79 +17,82 @@
 # limitations under the License.
 #
 
-# :data_bag is the object to search
-# :search_group is the groups name to search for, defaults to resource name
-# :group_name is the string name of the group to create, defaults to resource name
-# :group_id is the numeric id of the group to create, default is to allow the OS to pick next
-# :cookbook is the name of the cookbook that the authorized_keys template should be found in
-property :data_bag, String, default: 'users'
```

Comment:
https://docs.chef.io/release_notes_client/#deprecations-in-cookbook-resources

Reply:
Actually, you want `deprecated`:
```
property :thing2, String, deprecated: 'The thing2 property has been deprecated and will be removed in the next major release of this cookbook scheduled for 12/25/2018!'
```

Reply:
Yes indeed, didn't catch the data_bag bit first time

Reply:
But If I add that, then I should leave the functionality in right? It can't be deprecated and already not working right? Or is that okay?

Reply:
No you don't need to leave the functionality as it will do the following:
```
Starting Chef Infra Client, version 16.10.17
Patents: https://www.chef.io/patents
Creating a new client identity for default-centos-7 using the validator key.
resolving cookbooks for run list: ["users_test::default"]
Synchronizing Cookbooks:
  - users_test (1.7.1)
  - users (5.7.0)
Installing Cookbook Gems:
Compiling Cookbooks...
[2021-03-10T20:17:57+00:00] ERROR: Deprecated resource property used from /opt/kitchen/cache/cookbooks/users_test/recipes/default.rb:9:in `block in from_file'

  this has been deprecated!

Please consult the resource documentation for more information.

================================================================================
Recipe Compile Error in /opt/kitchen/cache/cookbooks/users_test/recipes/default.rb
================================================================================

Chef::Exceptions::DeprecatedFeatureError
----------------------------------------
Deprecated resource property used from /opt/kitchen/cache/cookbooks/users_test/recipes/default.rb:9:in `block in from_file'

  this has been deprecated!

Please consult the resource documentation for more information. (raising error due to treat_deprecation_warnings_as_errors being set)

Cookbook Trace: (most recent call first)
----------------------------------------
  /opt/kitchen/cache/cookbooks/users_test/recipes/default.rb:9:in `block in from_file'
  /opt/kitchen/cache/cookbooks/users_test/recipes/default.rb:6:in `from_file'
```

Also, it looks like we need to bump the required version of Chef to at least 13.9 or greater in `metadata.rb` since that feature wasn't included in older versions.

Reply:
I added the deprecated message and bumped the metadata to 14 as that is the lowest support version and it is quite close to 13.9.